recipes/full_finetune_distributed.py

```diff
@@ -266,6 +271,22 @@ def _setup_model(
         if self._dtype == torch.bfloat16:
             model = model.to(torch.bfloat16)
 
+        # We currently have two versions of activation checkpointing in this recipe
+        # for testing and BC purposes. ``enable_activation_checkpointing`` controls
+        # the older version of AC and this behavior is unchanged
+        # ac_mode and ac_option together control selective AC. This is only enabled
+        # when these are set AND ``enable_activation_checkpointing`` is set to False
+        # We'll clean this up as soon as testing of AC is complete
+        ac_mode = ac_mode
+        ac_option = ac_option
+
+        if (not enable_activation_checkpointing) and (ac_mode is not None):
```

Comment:
I understand why its this way but still reads a little weird. Like we have a check for `not enable_activation_checkpointing`, and underneath the check we apply AC (granted it is selective).